### ISSUE TITLE
plotting generates a blank window 

### ISSUE THREAD

**Issue by dav2017** [jheinen/GR.jl#124]
Something changed in the past weeks. When I plot data, now I get a blank window opens displaying no plot (but no error at REPL prompt). I tried different plot commands (plot, histogram), eg.:
```
using  GR
histogram(randn(1000)
```
I am running GR v. 0.31.0 and Julia v. 0.6.3 on Debian GNU/Linux.

**Comment by jheinen** [comment]
What's the output of the GR `version()` command?

**Comment by dav2017** [comment]
GR.version() returns "0.31.0".

A further information: I experienced the problem both using GR directly and as a backend of Plots (while Plots+pyplot is working as expected).

**Comment by jheinen** [comment]
Is the "blank" window a QtTerm? @FlorianRhiem : Can you reproduce this behaviour?

In any case, could please you try to:
```julia
ENV["GRDIR"]=""
Pkg.build("GR")
```

**Comment by dav2017** [comment]
The window is named GKS QtTerm.
After executing
```
ENV["GRDIR"]=""
Pkg.build("GR")
```
nothing changed (I closed and reopened the REPL and tried to plot something).

**Comment by jheinen** [comment]
... but you can create figures etc, right?
```julia
savefig("hist.pdf")
```

**Comment by dav2017** [comment]
Yes, savefig properly generates files (with graph drawn in them).

**Comment by dav2017** [comment]
Also, if I use GR within the Jupyter notebook the plot is correctly displayed.

**Comment by dav2017** [comment]
I did some more checking. The problem seems related to tiling window managers (dwm, i3): the window is blank if it is opened in tiled mode (and after that no window reshaping or making it float can make it usable for plotting). Instead, if the window is opened as a floating window it properly display the graph (and if I move to tiling mode it keeps working).
In the past it used not to be like that, I do not know what changed now, if my window manager configuration or the Qt term.

**Comment by jheinen** [comment]
Thanks for your hints. I will check this next week when I have access to a Debian machine - I'm currently on a conference.

**Comment by jheinen** [comment]
Works fine for me with macOS 10.14, Atom 1.28.2, GR master and Julia 0.6.4. Do you have a chance  to test it on a Mac?

![atom](https://user-images.githubusercontent.com/5718244/42870267-a403144e-8a77-11e8-8283-39823a31f68b.png)

**Comment by jheinen** [comment]
@dav2017 : unfortunately, I don't have access to a Debian box with a tiling WM. Will have to setup a VM for it ...

**Comment by korantir** [comment]
Well, I tried all rebuilds etc. Installed julia 0.6.4. All these methods didn't help. Although, for now I have error from issue nearby:  "GKS: svgplugin.dll: can't load library, error 126 (0x7e)" and previous solution does not help.
BTW on another win machine with latest atom, gr and julia 0.6.0 everything work properly. Sooo I'm quite sure that it is only my problem with GKS/GR build. I'll try to figure it out myself. Thanks for your quick answers @jheinen And for all your work too
P.S. Sadly I have no Mac.

**Comment by jheinen** [comment]
Could you please enter a command window (`cmd.exe`), change to the directory to where `svgplugin.dll` is located (`..\GR.jl\deps\gr\bin`) and invoke the following command:
```
dumpbin /dependents svgplugin.dll
```
![screen shot 2018-07-19 at 12 17 06](https://user-images.githubusercontent.com/5718244/42936877-b56336c0-8b4d-11e8-9ff9-238fc34cf823.png)

**Comment by korantir** [comment]
![cmd_2018-07-19_15-34-17](https://user-images.githubusercontent.com/8556803/42946558-d7a05da0-8b73-11e8-965b-33c1dae1fd9e.png)
I have feeling that something with path is wrong, but i don't have enough time to check code

for now path to svgplugin.dll is C:\Users\Administrator\.julia\v0.6\GR\deps\gr\bin

where Administrator is cyrillic, hope it doesn't matter.

**Comment by pkofod** [comment]
https://discourse.julialang.org/t/precompiling-plots-jl-fails-on-julia-1-0/13232/25

could this be related? read the last part of the discussion

**Comment by DoktorMike** [comment]
I'm the one who posted that question, and I'm also on a tiling WM (i3wm) on Ubuntu 17.10. I will try to run it using GNOME and see if it works there.

**Comment by RaulDurand** [comment]
> I can confirm that Plots and GR works on my machine when in Gnome. Thus for me it is an issue with my tiling window manager. I have no idea why though.

Same problem here with i3wm

**Comment by dav2017** [comment]
For i3wm a work-around is to open GR plot windows in floating mode. You can edit the i3 "config" file to automatically open this windows as floating. Add this line to the config file (which in Debian is in ~/.config/i3/):
  `for_window [class="gksqt"] floating enable `
which works both for GR and gr() via Plots.

**Comment by c42f** [comment]
The `for_window` workaround worked for me (on ubuntu 18.04 + i3wm). Some observations:

The `lib/gks/demo` application from GR itself shows the same blank window behavior without the `for_window` workaround, so this is unlikely related to GR.jl itself, this is probably a GR + tiling WM issue.

With GR.jl:

If the floating GR window is sent to the tiled layer, the plot resizes correctly. However, new plots in the tiled window are the size of the original floating window and the remaining part of the tiled window is not redrawn.

Also
1. If the `for_window` workaround is enabled
2. GR is started and a plot created
3. the `for_window` workaround is disabled
4. subsequent plots in now automatically tiled windows work ok.

**Comment by ctkelley** [comment]
I am having the same problem on a Mac with Julia 1.1. Everything was fine until I tried to use PlotlyJS.

**Comment by jheinen** [comment]
I have no idea why a PlotlyJS installation should affect GR.
So simple things like:
```julia
using  GR
histogram(randn(1000))
```
are no longer working after you added PlotlyJS?

**Comment by ctkelley** [comment]
I did a reinstall from scratch and the problem went away.

**Comment by floswald** [comment]
I'm still with this problem on Mac OS.  I rebuilt GR several times. thanks for any help!

```
➜  ~ julia
               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.5.3 (2020-11-09)
 _/ |\__'_|_|_|\__'_|  |  Official https://julialang.org/ release
|__/                   |

julia> using GR

julia> version()
"0.55.0.post5"

julia> histogram(randn(100))

julia> versioninfo()
Julia Version 1.5.3
Commit 788b2c77c1 (2020-11-09 13:37 UTC)
Platform Info:
  OS: macOS (x86_64-apple-darwin18.7.0)
  CPU: Intel(R) Core(TM) i7-4790K CPU @ 4.00GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-9.0.1 (ORCJIT, haswell)
Environment:
  JULIA_EDITOR = sublime

julia> 
```

<img width="940" alt="Screenshot 2021-03-03 at 10 46 26" src="https://user-images.githubusercontent.com/1021558/109786692-b7807400-7c0d-11eb-8b60-ddd0fd135333.png">

**Comment by floswald** [comment]
what's really weird is that in VScode that works totally fine.

<img width="1553" alt="Screenshot 2021-03-03 at 10 53 50" src="https://user-images.githubusercontent.com/1021558/109787692-c87db500-7c0e-11eb-8610-ea48556bb3a5.png">
